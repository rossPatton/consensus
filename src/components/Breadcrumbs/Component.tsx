import React, { memo } from 'react';
import { Link } from 'react-router-dom';

export const BreadcrumbsComponent = memo((props: any) => (
  <ul className="fs6 fw600 mB2 lh1 lsNone fx fxWrap brdB1 mB3 pB3">
    {console.log('props => ', props)}
    {props.country && (
      <li className="mR1">
        {!props.region && props.country.name}
        {props.region && (
          <Link to={`/${props.country.code}`}>
            {props.country.name}
          </Link>
        )}
      </li>
    )}
    {props.region && (
      <li className="mHide mR1">/</li>
    )}
    {props.region && (
      <li className="mR1">
        {!props.city && props.country.name}
        {props.city && (
          <Link to={`/${props.country.code}/${props.region.code}`}>
            {props.region.name}
          </Link>
        )}
      </li>
    )}
    <li className="mHide mR1">/</li>
    <li className="mR1">
      <Link to="/us/ny/new-york-city">
        New York City
      </Link>
    </li>
  </ul>
));
