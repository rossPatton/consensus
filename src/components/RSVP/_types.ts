import {History} from 'history';
import {RouteComponentProps} from 'react-router';

export type tSetRsvpOpts = {
  ev: React.FormEvent<HTMLFormElement> | React.MouseEvent<HTMLButtonElement>,
  meetingId: number,
}

export type tStore = {
  rsvps: ts.thunk<ts.rsvp[]>,
  session: ts.thunk<ts.session>,
};

export type tState = {
  hasMounted: boolean,
  rsvp?: ts.rsvp, // pick out the rsvp from the rsvps state object
};

type tProps = {
  className?: string,
  meeting: ts.meeting,
  role?: ts.role,
  rsvp?: ts.rsvp,
  session: ts.session,
};

export type tContainerProps = RouteComponentProps<any> & tProps & {
  history: History,
  patchRsvpDispatch: (query: ts.rsvpQuery) => ts.thunkPayload<ts.rsvp>,
  postRsvpDispatch: (query: ts.rsvpQuery) => ts.thunkPayload<ts.rsvp>,
  rsvps: ts.rsvp[],
};

export type tComponentProps = ts.mediaContext & tProps & {
  hasMounted: boolean,
  setRsvp: (opts: tSetRsvpOpts) => void,
};
