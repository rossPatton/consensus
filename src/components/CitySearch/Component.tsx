import _ from 'lodash';
import React, {memo} from 'react';

import stateCodeMap from '../../json/usa/stateCodeMap.json';
import {tComponentProps} from './_types';

export const CitySearchComponent = memo((props: tComponentProps) => {
  const onCityChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const city = _.find(props.cities, c => c.name === ev.currentTarget.value);
    if (city) {
      props.updateState({
        city: city.name,
        cityId: city.id,
        region: city.region,
        regionId: city.regionId,
      });
    }
  };

  const onRegionChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const region = ev.currentTarget.value;
    if (region !== props.region) {
      props.updateState({
        city: '',
        cityId: 0,
        region,
        regionId: 0,
      });
    }
  };

  return (
    <>
      {props.showRegionField && (
        <>
          <h2 className="fs5 mB1 lh1">
            Pick a different state
          </h2>
          <label htmlFor="stateSelect">
            <select
              name="type"
              id="stateSelect"
              className="mB4 row bgWhite"
              value={props.region}
              onBlur={onRegionChange}
              onChange={onRegionChange}>
              {Object.keys(stateCodeMap).map(state => (
                <option key={state} value={state}>
                  {state}
                </option>
              ))}
            </select>
          </label>
        </>
      )}
      <h2 className="fs5 mB1 lh1">
        City in <span className="dInBl mR2">
          {(!props.showRegionField && props.geo.region) && props.geo.region}
          {(props.showRegionField && props.region) && props.region}
        </span>
        {!props.showRegionField && (
          <button
            type="button"
            onClick={props.toggleRegionField}>
            Not the right state?
          </button>
        )}
      </h2>
      {props.label && (
        <p className="fs5 copyBlack mB1">
          {props.label}
        </p>
      )}
      <div className="mB4">
        {props.cityId === 0 && (
          <label htmlFor="citySelect">
            <select
              name="type"
              id="citySelect"
              className="row bgWhite"
              value={props.city}
              onBlur={onCityChange}
              onChange={onCityChange}>
              {props.cities.map((city: tCity) => (
                <option key={city.id} value={city.name}>
                  {city.name}, {city.region}
                </option>
              ))}
            </select>
          </label>
        )}
        {props.cityId > 0 && (
          <div className="brdA1 p3 black br8 dInBl">
            <b>{props.city}</b>, <span className="dInBl mR3">{props.region}</span>
            <button
              type="button"
              onClick={() => {
                props.updateState({
                  city: '',
                  cityId: 0,
                });
              }}>
              Not the right city?
            </button>
          </div>
        )}
      </div>
    </>
  );
});
