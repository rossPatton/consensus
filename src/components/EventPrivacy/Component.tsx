import React, {memo} from 'react';

import {tProps} from './_types';

export const EventPrivacyComponent = memo((props: tProps) => (
  <small className="bgYellowLite br8 p1 pL2 pR2 mR2">
    {props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Lock Emoji">
          ðŸ”’
        </span>
        Private Event
      </>
    )}
    {!props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Tada Emoji">
          ðŸŽ‰
        </span>
        Public Event
      </>
    )}
  </small>
));
