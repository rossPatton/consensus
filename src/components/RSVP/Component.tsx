import cx from 'classnames';
import _ from 'lodash';
import React from 'react';

import {tComponentProps} from './_types';

export const RSVPComponent = (props: tComponentProps) => {
  const {meeting, rsvp, session, setRsvp} = props;
  const {id: meetingId} = meeting;
  const {profile = {}} = session;
  const {privateRSVP: userRSVPsPrivately} = profile as ts.user;
  const method = typeof rsvp === 'undefined' ? 'POST' : 'PATCH';

  return (
    <form
      method={method}
      onSubmit={ev => setRsvp({ev, meetingId})}
      className={cx({
        [props.className]: typeof props.className === 'string',
        'absolute t-hdr l r pt-2 pb-2 bg-gray-2': props.isMobile,
      })}>
      <fieldset>
        <div className="text-center d:text-left d:flex d:items-center text-sm leading-none">
          <legend className="block font-semibold mb-1 mr-2 d:mb-0">
            {rsvp?.value
              ? 'Change RSVP'
              : `RSVP ${userRSVPsPrivately ? 'Privately' : 'Publicly'}`}
          </legend>
          <button
            value="yes"
            type={props.hasMounted ? 'button' : 'submit'}
            onClick={ev => setRsvp({ev, meetingId})}
            className={cx({
              'p-2 pl-3 pr-3 mr-1 hover:bg-green-1': true,
              'bg-green-1': rsvp?.value === 'yes',
            })}>
            Yes
          </button>
          <button
            value="no"
            type={props.hasMounted ? 'button' : 'submit'}
            onClick={ev => setRsvp({ev, meetingId})}
            className={cx({
              'p-2 pl-3 pr-3 mr-1 hover:bg-red-1': true,
              'bg-red-1': rsvp?.value === 'no',
            })}>
            No
          </button>
          <button
            value="maybe"
            type={props.hasMounted ? 'button' : 'submit'}
            onClick={ev => setRsvp({ev, meetingId})}
            className={cx({
              'p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1': true,
              'bg-yellow-1': rsvp?.value === 'maybe',
            })}>
            Maybe
          </button>
        </div>
      </fieldset>
    </form>
  );
};
