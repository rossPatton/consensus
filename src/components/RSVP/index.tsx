import _ from 'lodash';
import loglevel from 'loglevel';
import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';

import {MediaContext} from '~app/context/MatchMediaProvider/_context';
import {patchRsvps, postRsvps} from '~app/redux';

import {tContainerProps, tSetRsvpOpts, tState, tStore} from './_types';
import {RSVPComponent} from './Component';

class RSVPContainer extends PureComponent<tContainerProps, tState> {
  static contextType = MediaContext;

  constructor(props: tContainerProps) {
    super(props);
    const {rsvps} = props;
    const rsvp = _.find(rsvps, rsvp => rsvp.eventId === this.props.event.id);

    // only purpose for this state, is to update the UI right away
    const hasRSVPed = !!rsvp && rsvp.value !== null;

    this.state = {
      hasRSVPed,
      hasMounted: false,
      rsvp,
    };
  }

  componentDidMount() {
    this.setState({
      hasMounted: true,
    });
  }

  // post or patch, depending on RSVP status
  setRsvp = async (opts: tSetRsvpOpts) => {
    opts.ev.preventDefault();

    const {rsvp} = this.state;
    const {history, patchRsvpDispatch, postRsvpDispatch, session} = this.props;
    const {profile = {}} = session;
    const {privateRSVP: userRSVPsPrivately = true} = profile as tUser;

    if (!session.isAuthenticated) return history.push('/login');

    const method = typeof rsvp === 'undefined' ? 'POST' : 'PATCH';
    const dispatch = method === 'PATCH' ? patchRsvpDispatch : postRsvpDispatch;
    const hasRSVPed = !this.state.hasRSVPed;
    this.setState({hasRSVPed});

    let {value} = opts.ev.currentTarget;
    if (method === 'PATCH') {
      // if changing rsvp, use new value.
      // else if removing rsvp (clicking same option twice), set to null
      value = opts.ev.currentTarget.value !== rsvp.value
        ? opts.ev.currentTarget.value
        : '';
    }

    try {
      dispatch({
        eventId: opts.eventId,
        type: userRSVPsPrivately ? 'private' : 'public',
        value,
      })
        .then(res => {
          return this.setState({
            rsvp: res.payload,
          });
        })
        .catch(loglevel.error);
    } catch (err) {
      return loglevel.error(err);
    }
  }

  render() {
    if (this.props.session.type === 'org') return null;
    const {isMobile, isDesktop} = this.context;

    return (
      <RSVPComponent
        className={this.props.className}
        event={this.props.event}
        hasMounted={this.state.hasMounted}
        hasRSVPed={this.state.hasRSVPed}
        isDesktop={isDesktop}
        isMobile={isMobile}
        rsvp={this.state.rsvp}
        session={this.props.session}
        setRsvp={this.setRsvp}
      />
    );
  }
}

const mapStateToProps = (store: tStore) => ({
  isLoading: store.rsvps.isLoading,
  rsvps: store.rsvps.data,
  session: store.session.data,
});

const mapDispatchToProps = (dispatch: Function) => ({
  patchRsvpDispatch: (query: tRSVPQuery) => dispatch(patchRsvps(query)),
  postRsvpDispatch: (query: tRSVPQuery) => dispatch(postRsvps(query)),
});

const RSVP = connect(mapStateToProps, mapDispatchToProps)(RSVPContainer);
export default withRouter(RSVP);
