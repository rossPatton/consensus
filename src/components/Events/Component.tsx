import cx from 'classnames';
import dayJS from 'dayjs';
import _ from 'lodash';
import qs from 'query-string';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {ExternalLink, PlaceholderImage, RSVP} from '../../components';
import {objToQueryString} from '../../utils';
import {tComponentProps} from './_types';

export const EventsComponent = memo((props: tComponentProps) => (
  <ul
    className={cx({
      'fx fxdRow taL': props.horizontal,
      'jcBetween': props.horizontal && props.events.length === 4,
    })}>
    {props.events.map((ev, i) => {
      const isPastEvent = dayJS(ev.date).isBefore(dayJS());

      return (
        <li
          key={ev.id}
          className={cx({
            'mB2': !props.horizontal,
            'col row': props.horizontal,
            'mR4': props.horizontal && i !== props.events.length - 1,
          })}>
          <div
            className={cx({
              'fx aiCtr': !props.horizontal,
              'p3 hvrBgGrey1 br8': props.isEditable,
            })}>
            <div
              className={cx({
                'mB2 bgWhite': props.horizontal,
                'mR3 bgGrey2': !props.horizontal,
              })}>
              <Link
                className="noUnderline"
                to={ev.isDraft
                  ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                  : `/event/${ev.id}`}>
                <PlaceholderImage
                  height={100}
                  seed={ev.id}
                  width={200}
                />
              </Link>
            </div>
            <div>
              <div
                className={cx({
                  'fx aiCtr mB1 fs7 fw600 lh1': true,
                  o5: isPastEvent,
                })}>
                <time className="mR1" dateTime={ev.date}>
                  {dayJS(ev.date).format('MMM DD | h:mmA')}
                </time>
                {!props.horizontal && (
                  <>
                    <span className="mR1">@</span>
                    {ev.locationLink && (
                      <ExternalLink
                        noFollow
                        className="mR1"
                        to={ev.locationLink}>
                        {ev.location}
                      </ExternalLink>
                    )}
                  </>
                )}
                {!ev.locationLink && ev.location}
              </div>
              <h3
                className={cx({
                  'fx aiCtr ttCap': true,
                  mB1: props.isEditable,
                  fs4: props.horizontal,
                })}>
                <Link
                  className="noUnderline"
                  to={ev.isDraft
                    ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                    : `/event/${ev.id}`}>
                  {ev.title}
                </Link>
              </h3>
              {props.showRSVPs && (
                <div
                  className={cx({
                    'fx aiCtr fs6 lh1': true,
                    hide: props.horizontal,
                    o5: isPastEvent,
                  })}>
                  {!ev.isDraft && <RSVP event={ev} />}
                  {ev.isDraft && 'This event is not published yet. Click to edit.'}
                </div>
              )}
              {props.showOrgName
                && (
                  <Link
                    to={`/org/${ev.orgId}`}
                    className={cx({
                      'fw600 fs6 lh1': true,
                      o5: isPastEvent,
                    })}>
                    {ev.orgName}
                  </Link>
                )}
              {!isPastEvent
              && props.isEditable
              && (
                <div className="fx aiCtr mB1">
                  <div className="bgGrey4 br4 fs7 lh1 mR2 p1 pL2 pR2 white">
                    {ev.isDraft ? 'Draft' : 'Published'}
                  </div>
                  <div className="fx aiCtr mR1">
                    <Link
                      to={`/org/${ev.orgId}/planMeeting?${objToQueryString(ev)}`}
                      className="btn fs7 fw600 hvrBgGrey1 lh1 noUnderline p1 pL2 pR2">
                      <span
                        role="img"
                        aria-label="Hand with Pen Emoji">
                        ✍️
                      </span>
                      Edit
                    </Link>
                  </div>
                  <button
                    onClick={e => props.deleteEvent(e, ev.id)}
                    className="aiCtr fs7 fw600 fx hvrBgGrey1 lh1">
                    <span
                      role="img"
                      className="mR1"
                      aria-label="Big X Emoji">
                      ✖️
                    </span>
                    Delete
                  </button>
                </div>
              )}
            </div>
          </div>
        </li>
      );
    })}
  </ul>
));
