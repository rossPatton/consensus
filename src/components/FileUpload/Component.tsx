import cx from 'classnames';
import React, {memo} from 'react';
import {v1} from 'uuid';

import {Emoji} from '~app/components';

import {tComponentProps} from './_types';

const baseUrl = 'https://consensus.nyc3.digitaloceanspaces.com';

export const FileUploadComponent = memo((props: tComponentProps) => (
  <>
    <h3 className="text-base mb-1">
      {props.title}
    </h3>
    <div
      id="imagePreview"
      className={cx({
        'animated bg-gray-1 p-3 mb-3 flex flex-col items-center': true,
        'fadeInDown': props.img,
        'hidden': !props.img,
      })}>
      <button
        type="button"
        onClick={props.removeImage}
        className="mb-2 p-2">
        <Emoji
          label="Close Emoji"
          emoji="✖️"
        />
        Remove Image
      </button>
      {props.img && (
        <img
          alt="Your uploaded file preview."
          height={props.width}
          width={props.width}
          src={`${baseUrl}/${props.folder}/${props.img}${__DEV__ ? `?v=${v1()}` : ''}`}
          className={cx({
            'rounded w-full': !props.fieldKey.includes('Avatar'),
            'rounded-circ': props.fieldKey.includes('Avatar'),
          })}
        />
      )}
    </div>
    <label
      htmlFor="fileUpload"
      className={cx({
        'fadeInDown bg-gray-1 border cursor-ptr flex flex-col items-center mb-3 p-3 rounded animated': true,
        'fadeInDown': !props.img,
        'hidden': props.img,
      })}>
      <input
        accept="image/png, image/jpeg, image/gif"
        id="fileUpload"
        name="eventFeaturedImage"
        onChange={props.setImage}
        type="file"
        className={cx({
          hidden: true,
        })}
      />
      <span className="btn flex items-center mb-1 p-1">
        <Emoji
          label="Picture (Camera) Emoji"
          emoji="📷"
        />
        Upload Your Image
      </span>
      {props.info && (
        <small>{props.info}</small>
      )}
    </label>
  </>
));
