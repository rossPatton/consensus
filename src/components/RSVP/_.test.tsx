import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import render from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import {testMeeting2, testUserSession, testUserSession2} from '~app/constants/jest';
import MatchMediaProvider from '~app/context/MatchMediaProvider';

import RSVP from '.';
import {RSVPComponent} from './Component';

const mockStore = configureStore();

describe('components/RSVP', () => {
  let store = mockStore({});

  beforeEach(() => {
    store = mockStore({
      rsvps: {
        data: [{
          meetingId: 1,
          type: 'public',
          userId: 100,
          value: 'yes',
        }],
      },
      session: testUserSession,
    });
  });

  it('renders mobile without crashing', () => {
    const meeting = {id: 1} as ts.meeting;

    const component = render.create((
      <Provider store={store}>
        <MemoryRouter>
          <MatchMediaProvider isMobile>
            <RSVP meeting={meeting} />
          </MatchMediaProvider>
        </MemoryRouter>
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders desktop without crashing', () => {
    const meeting = {id: 1} as ts.meeting;

    const component = render.create((
      <Provider store={store}>
        <MemoryRouter>
          <MatchMediaProvider isDesktop>
            <RSVP meeting={meeting} />
          </MatchMediaProvider>
        </MemoryRouter>
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders null if group type', () => {
    const meeting = {id: 1} as ts.meeting;
    const customStore = mockStore({
      rsvps: {
        data: [{
          meetingId: 1,
          type: 'public',
          userId: 100,
          value: undefined,
        }],
      },
      session: {
        data: {
          isAuthenticated: false,
          type: 'group',
        },
      },
    });

    const component = render.create((
      <Provider store={customStore}>
        <MemoryRouter>
          <MatchMediaProvider isDesktop>
            <RSVP meeting={meeting} />
          </MatchMediaProvider>
        </MemoryRouter>
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component renders', () => {
    const component = render.create((
      <RSVPComponent
        isDesktop
        isMobile={false}
        meeting={testMeeting2}
        // @ts-ignore
        session={testUserSession}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component renders with rsvp', () => {
    const component = render.create((
      <RSVPComponent
        isDesktop
        isMobile={false}
        meeting={testMeeting2}
        rsvp={{
          meetingId: 1,
          type: 'public',
          userId: 100,
          value: 'yes',
        }}
        // @ts-ignore
        session={testUserSession}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component renders with maybe rsvp', () => {
    const component = render.create((
      <RSVPComponent
        isDesktop
        isMobile={false}
        meeting={testMeeting2}
        rsvp={{
          meetingId: 1,
          type: 'public',
          userId: 100,
          value: 'maybe',
        }}
        // @ts-ignore
        session={testUserSession2}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component onSubmit', () => {
    const component = render.create((
      <RSVPComponent
        isDesktop
        isMobile={false}
        hasMounted
        meeting={testMeeting2}
        // @ts-ignore
        session={testUserSession2}
        setRsvp={jest.fn()}
      />
    ));

    component.root.findByType('form').props.onSubmit();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component click rsvp buttons', () => {
    const component = render.create((
      <RSVPComponent
        isDesktop
        isMobile={false}
        hasMounted
        meeting={testMeeting2}
        // @ts-ignore
        session={testUserSession}
        setRsvp={jest.fn()}
      />
    ));

    component.root.findByProps({value: 'yes'}).props.onClick();
    component.root.findByProps({value: 'maybe'}).props.onClick();
    component.root.findByProps({value: 'no'}).props.onClick();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
