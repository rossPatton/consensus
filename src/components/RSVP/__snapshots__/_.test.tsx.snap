// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RSVP component click rsvp buttons 1`] = `
<form
  className=""
  method="POST"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        RSVP Publicly
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1"
        onClick={[Function]}
        type="button"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="button"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1"
        onClick={[Function]}
        type="button"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP component onSubmit 1`] = `
<form
  className=""
  method="POST"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        RSVP Publicly
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1"
        onClick={[Function]}
        type="button"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="button"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1"
        onClick={[Function]}
        type="button"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP component renders 1`] = `
<form
  className=""
  method="POST"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        RSVP Publicly
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1"
        onClick={[Function]}
        type="submit"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="submit"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1"
        onClick={[Function]}
        type="submit"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP component renders with maybe rsvp 1`] = `
<form
  className=""
  method="PATCH"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        Change RSVP
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1"
        onClick={[Function]}
        type="submit"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="submit"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1 bg-yellow-1"
        onClick={[Function]}
        type="submit"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP component renders with rsvp 1`] = `
<form
  className=""
  method="PATCH"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        Change RSVP
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1 bg-green-1"
        onClick={[Function]}
        type="submit"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="submit"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1"
        onClick={[Function]}
        type="submit"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP renders desktop without crashing 1`] = `
<form
  className=""
  method="PATCH"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        Change RSVP
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1 bg-green-1"
        onClick={[Function]}
        type="button"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="button"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1"
        onClick={[Function]}
        type="button"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP renders mobile without crashing 1`] = `
<form
  className=""
  method="PATCH"
  onSubmit={[Function]}
>
  <fieldset>
    <div
      className="text-center d:text-left d:flex d:items-center text-sm leading-none"
    >
      <legend
        className="block font-semibold mb-1 mr-2 d:mb-0"
      >
        Change RSVP
      </legend>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-green-1 bg-green-1"
        onClick={[Function]}
        type="button"
        value="yes"
      >
        Yes
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-red-1"
        onClick={[Function]}
        type="button"
        value="no"
      >
        No
      </button>
      <button
        className="p-2 pl-3 pr-3 mr-1 hover:bg-yellow-1"
        onClick={[Function]}
        type="button"
        value="maybe"
      >
        Maybe
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`components/RSVP renders null if group type 1`] = `null`;
